### ISSUE TITLE
Legacy importing type from namespace is not supported

### ISSUE THREAD

**Issue by JounQin** [esbuild-kit/tsx#83]
### Bug description

```ts
import { Rule, Scope } from 'eslint'; // `Rule` and `Scope` are both ts namespace
import Variable = Scope.Variable;
import CodePathSegment = Rule.CodePathSegment;
```

```log
import Variable = Scope.Variable;
                        ^

TypeError: Cannot read properties of undefined (reading 'Variable')
```

### Reproduction

```ts
import { Rule, Scope } from 'eslint';
import Variable = Scope.Variable;
import CodePathSegment = Rule.CodePathSegment;
```

### Environment

```shell
  System:
    OS: macOS 13.0
    CPU: (10) arm64 Apple M1 Max
    Memory: 31.99 GB / 64.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.16.0 - ~/Library/Caches/fnm_multishells/55951_1660375587006/bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn
    npm: 8.16.0 - ~/Library/Caches/fnm_multishells/55951_1660375587006/bin/npm
  npmPackages:
    tsx: ^3.8.2 => 3.8.2
```

### Can you contribute a fix?

- [X] I’m interested in opening a pull request for this issue.

**Comment by privatenumber** [comment]
```diff
- import { Rule, Scope } from https://github.com/eslint/eslint; // `Rule` and `Scope` are both ts namespace
+ import type { Rule, Scope } from https://github.com/eslint/eslint; // `Rule` and `Scope` are both ts namespace
```

```diff
> import { Rule, Scope } from 'eslint'; // `Rule` and `Scope` are both ts namespace
> import Variable = Scope.Variable;
> import CodePathSegment = Rule.CodePathSegment;
> ```
>

**Comment by JounQin** [comment]
I know how to workaround this, I just forgot that could be a esbuild issue. I'll raise a new issue there instead. (Stupid me. 😅)

**Comment by privatenumber** [comment]
I haven't tested yet but I think this is unblocked as of:
- https://github.com/esbuild-kit/core-utils/releases/tag/v2.2.0
- https://github.com/evanw/esbuild/releases/tag/v0.15.4

**Comment by privatenumber** [comment]
I tested it out just now. Some observations:

- Tree-shaking must be enabled to get this benefit. [Without tree-shaking.](https://hyrious.me/esbuild-repl/?version=0.15.5&mode=transform&input=import+%7B+someNamespace+%7D+from+%27.%2Fsome-file%27%0Aimport+bar+%3D+someNamespace.foo%3B%0A%0Aconst+a%3A+bar+%3D+1%3B%0Aconsole.log%28a%29%3B%0A&options=--loader%3Dts) [With tree -shaking.](https://hyrious.me/esbuild-repl/?version=0.15.5&mode=transform&input=import+%7B+someNamespace+%7D+from+%27.%2Fsome-file%27%0Aimport+bar+%3D+someNamespace.foo%3B%0A%0Aconst+a%3A+bar+%3D+1%3B%0Aconsole.log%28a%29%3B%0A&options=--loader%3Dts+--tree-shaking) I haven't considered enabling tree-shaking but I'll probably have to think about it a little more; I wonder if it is safe.
- In  ESM mode (`package.json#type:module`), the import statements are preserved, so it fails on the first namespace import because it's preserved for ESM:
```ts
import { Rule, Scope } from https://github.com/eslint/eslint;
```

eg.
```
SyntaxError: The requested module './some-file' does not provide an export named 'someNamespace'
```

**Comment by JounQin** [comment]
> Tree-shaking must be enabled to get this benefit

Or can we add a flag to enable it manually?

**Comment by JounQin** [comment]
> Tree-shaking must be enabled to get this benefit

Or can we add a flag to enable it manually?

**Comment by JounQin** [comment]
> Tree-shaking must be enabled to get this benefit

Or can we add a flag to enable it manually?
